src/tests/OracleWrapper.t.sol

```diff
@@ -17,52 +17,65 @@
 pragma solidity ^0.8.16;
 
 import "forge-std/Test.sol";
+
+import { DssInstance, MCD } from "dss-test/MCD.sol";
+import { FlapperDeploy } from "src/deploy/FlapperDeploy.sol";
+import { FlapperInit } from "src/deploy/FlapperInit.sol";
 import { OracleWrapper } from "src/OracleWrapper.sol";
 
-interface PipLike {
-    function read() external view returns (bytes32);
+interface ChainlogLike {
+    function getAddress(bytes32) external view returns (address);
 }
 
-contract MockMedianizer {
-    uint256 public price;
-    bool    public has = true;
-
-    function setPrice(uint256 _price) external {
-        price = _price;
-    }
-
-    function setHas(bool _has) external {
-        has = _has;
-    }
+interface OsmLike {
+    function src() external view returns (address);
+}
 
-    function read() external view returns (bytes32) {
-        require(has);
-        return bytes32(price);
-    }
+interface PipLike {
+    function read() external view returns (bytes32);
+    function kiss(address) external;
+    function diss(address) external;
 }
 
 contract OracleWrapperTest is Test {
-    MockMedianizer public medianizer;
-    PipLike        public oracleWrapper;
+    PipLike public medianizer;
+    PipLike public oracleWrapper;
+    uint256 public medianizerPrice;
+
+    address constant LOG = 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F;
+
+    address immutable PAUSE_PROXY = ChainlogLike(LOG).getAddress("MCD_PAUSE_PROXY");
+    address immutable PIP_ETH     = ChainlogLike(LOG).getAddress("PIP_ETH");
 
     uint256 constant WAD = 10 ** 18;
 
     function setUp() public {
-        medianizer = new MockMedianizer();
-        medianizer.setPrice(727 * WAD);
-        medianizer.setHas(true);
+        medianizer = PipLike(OsmLike(PIP_ETH).src());
+
+        // Get current price
+        vm.prank(PAUSE_PROXY);
+        medianizer.kiss(address(this));
+        medianizerPrice = uint256(medianizer.read());
+        assertGt(medianizerPrice, 0);
+        vm.prank(PAUSE_PROXY);
+        medianizer.diss(address(this));
 
-        oracleWrapper = PipLike(address(new OracleWrapper(address(medianizer), address(this), 1800)));
+        oracleWrapper = PipLike(FlapperDeploy.deployOracleWrapper(address(medianizer), address(this), 1800));
```

Comment:
Shall we use 1200 to match the likely value that will be used in the spell?

Reply:
I think it's find to have it different. Up to Maker governance to decide.